integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala

```diff
@@ -916,6 +930,19 @@ object AlterTableUtil {
         }
       }
     }
+    // should not be present in index tables
+    val indexTables = CarbonIndexUtil.getSecondaryIndexes(carbonTable)
+    if (!indexTables.isEmpty) {
+      indexTables.asScala.map(indexTable => {
+        CarbonEnv.getCarbonTable(Some(carbonTable.getDatabaseName), indexTable)(sparkSession)
+          .getCreateOrderColumn.asScala.map(column => {
+          if (longStringCols.contains(column.getColName)) {
+            throw new MalformedCarbonCommandException(s"col ${column.getColName} is " +
```

Comment:
Can change to, "Cannot Alter column 'b' to Long_string_column, as the column exists in a secondary index with name `indexTable`.  LONG_STRING_COLUMNS is not allowed on secondary index."

Reply:
done